galleries/examples/images_contours_and_fields/image_exact_placement.py

```diff
@@ -0,0 +1,131 @@
+"""
+=========================================
+Placing images, preserving relative sizes
+=========================================
+
+By default Matplotlib resamples images created with `~.Axes.imshow` to
+fit inside the parent `~.Axes`.  This can mean that images that have very
+different original sizes can end up appearing similar in size.
+
+Sometimes, however,  it is desirable to keep the images the same relative size, or
+even to make the images keep exactly the same pixels as the original data.
+Matplotlib does not automatically make either of these things happen,
+but it is possible with some manual manipulation.
+
+Preserving relative sizes
+=========================
+
+If the size of the images are amenable, we can preserve the relative sizes of two
+images by using either the *width_ratio* or *height_ratio* of the subplots.  Which
+one you use depends on the shape of the image and the size of the figure.
+
+By default the two images are made a similar size, despite one being 1.5 times the width
+of the other:
+"""
+
+# sphinx_gallery_thumbnail_number = -1
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import matplotlib.patches as mpatches
+
+# make the data:
+N = 450
+x = np.arange(N) / N
+y = np.arange(N) / N
+
+X, Y = np.meshgrid(x, y)
+R = np.sqrt(X**2 + Y**2)
+f0 = 5
+k = 100
+a = np.sin(np.pi * 2 * (f0 * R + k * R**2 / 2))
+A = a[:100, :300]
+B = A[:40, :200]
+
+# plot with default axes handling:
+fig, axs = plt.subplots(1, 2, facecolor='aliceblue')
+
+axs[0].imshow(A, vmin=-1, vmax=1)
+axs[1].imshow(B, vmin=-1, vmax=1)
+rect = mpatches.Rectangle((0, 0), 200, 40, linewidth=1, edgecolor='r', facecolor='none')
+axs[0].add_patch(rect)
+
+# %%
+# Note that both images are rendered at a 1:1 ratio, but are made to look almost the
+# same width, despite image B being smaller than image A.
```

Comment:
```suggestion
# Note that both images have an aspect ratio of 1 (i.e. pixels are square), but
# pixels sizes differ because the images are scaled to the same width.
```

Optional suggestion: Use significantly lower resolution so that one can see the individual pixels.

Reply:
Can we do that with sphinx-gallery?

Reply:
What I meant has nothing to do with sphinx-gallery: Use N=50 or 100 instead of 450, and equally scale down your image slices. You may need to adjust you formula for `a` to have a reasonably slowly varying function so that it does not change too much between pixels on the coarse pixel grid.

Reply:
So make the images small when I do the explicit placement below? Certainly they could be smaller, but maybe not too much smaller?